Mathlib/Order/Fin/Basic.lean

```diff
@@ -170,6 +170,16 @@ lemma predAbove_left_monotone (i : Fin (n + 1)) : Monotone fun p ↦ predAbove p
 @[simps!] def predAboveOrderHom (p : Fin n) : Fin (n + 1) →o Fin n :=
   ⟨p.predAbove, p.predAbove_right_monotone⟩
 
+lemma orderHom_injective_iff {α : Type*} [PartialOrder α] {n : ℕ} (f : Fin (n + 1) →o α) :
```

Comment:
@YaelDillies it looks like this one might have a generalisation in terms of succ orders: do you know if that exists already?

Reply:
No it doesn't. It would belong after `strictMono_iff_lt_succ`

Reply:
I have moved the lemma upwards in the file.